app/assets/javascripts/components/plots/forms/plot_form.jsx

```diff
@@ -0,0 +1,348 @@
+          label: 'heatmap'
+        }
+      ];
+
+      return (
+        <Select
+          hasNull={false}
+          label={'Plot Type'}
+          onChange={this.props.changePlotType}
+          options={options}
+          value={this.props.plot.plotType}
+        />
+      );
+    }
+
+    manifestSelector() {
+      const options = this.props.manifests.map(({ id, name }) => {
+        return {
+          value: id,
+          label: name
+        };
+      });
+
+      return (
+        <Select
+          hasNull={false}
+          label={'Manifest'}
+          onChange={(id) => this.props.selectManifest(id)}
+          options={options}
+          value={this.props.plot.manifestId}
+        />
+      );
+    }
+
+    formFields() {
+      const injectedProps = {
+        updatePlot: this.props.updatePlot,
+        plot: this.props.plot,
+        consignment: this.props.consignment
+      };
+
+      return plotFormFields.map((Field, i) => <Field key={i} {...injectedProps} />);
+    }
+
+    //when a new series is added to the data array add an id to the series and update plot configuration
+    addDataRef() {
+      const newSeries = {
+        ...this.state.data,
+        id: Math.random()
+      };
+      let updatedPlot = { ...this.props.plot };
+      updatedPlot.data = addSeriesData(newSeries, updatedPlot.data);
+      this.props.updatePlot(updatedPlot);
+    }
+
+    //delete a series from the data array by id
+    removePlotRefData(id) {
+      let updatedPlot = { ...this.props.plot };
+      updatedPlot.data = updatedPlot.data.filter(series => series.id != id);
+      this.props.updatePlot(updatedPlot);
+    }
+
+    //form fields are injected with props
+    dataFormFields() {
+      const injectedProps = {
+        updatePlot: (data) => this.setState({ data }),
+        plot: this.state.data,
+        consignment: this.props.consignment
+      };
+
+      return dataFormFields.map((Field, i) => <Field key={i} {...injectedProps} />);
+    }
+
+    dataRefList() {
+      const data = this.props.plot.data || [];
+
+      return (
+        <ul>
+          {data.map(({ name, id }) => (
+            <li key={id}>
+              {name + ' '}
+              <i className='fa fa-times' aria-hidden='true'
+                 onClick={() => this.removePlotRefData(id)}>
+              </i>
+            </li>
+          ))}
+        </ul>
+      );
+    }
+
+    render () {
+      const { toggleEditing, selectedManifest, handleSave, isNewPlot, plot, consignment } = this.props;
+
+      return (
+        <div className='plot-form-container'>
+          <ButtonBar
+            className='actions'
+            buttons={[
+              {
+                click: handleSave,
+                icon: 'floppy-o',
+                label: 'save'
+              }, {
+                click: () => toggleEditing(false),
+                icon: 'ban',
+                label: 'cancel'
+              }
+            ]}
+          />
+          {isNewPlot ? (
```

Comment:
This plotType and manifest selection stuff seems like it should be its own component, since you need to set it before this PlotForm can be drawn.